common/models/user.js

```diff
@@ -354,6 +354,60 @@ module.exports = function(User) {
   };
 
   /**
+   * Change this user's password.
+   *
+   * @param {*} userId Id of the user changing the password
+   * @param {string} oldPassword Current password, required in order
+   *   to strongly verify the identity of the requesting user
+   * @param {string} newPassword The new password to use.
+   * @param {object} [options]
+   * @callback {Function} callback
+   * @param {Error} err Error object
+   * @promise
+   */
+  User.changePassword = function(userId, oldPassword, newPassword, options, cb) {
+    if (cb === undefined && typeof options === 'function') {
+      cb = options;
+      options = undefined;
+    }
+    cb = cb || utils.createPromiseCallback();
+
+    // Make sure to use the constructor of the (sub)class
+    // where the method is invoked from
+    const Ctor = this;
+
+    Ctor.findById(userId, options, function(err, inst) {
+      if (err) return cb(err);
+
+      if (!inst) {
+        const err = new Error(`User ${userId} not found`);
+        Object.assign(err, {
+          code: 'USER_NOT_FOUND',
+          statusCode: 400,
```

Comment:
Good catch. I was thinking about 404 too, but it seemed inappropriate to me.

> The server has not found anything matching the Request-URI.

The Request URI is `/api/users/change-password`, which is a valid existing path.

The problem is in the access token supplied in the Authorization header, because the user authenticated by this token no longer exists.

On the second thought, I agree with @superkhau that 401 is a more appropriate response status code.